sql/parser/builtins.go

```diff
@@ -311,10 +312,19 @@ var builtins = map[string][]builtin{
 		builtin{
 			types:      typeList{},
 			returnType: DummyFloat,
+			impure:     true,
 			fn: func(args DTuple) (Datum, error) {
 				return DFloat(rand.Float64()), nil
 			},
 		},
+		builtin{
+			types:      typeList{intType},
+			returnType: DummyFloat,
+			impure:     true,
+			fn: func(args DTuple) (Datum, error) {
+				return DFloat(rand.New(rand.NewSource(int64(args[0].(DInt)))).Float64()), nil
```

Comment:
Won't this produce the same random number for every row? (we need a test that `order by random(5)` produces different results from `order by random(6)`) `random(seed)` needs to construct the RNG once per query and then evaluate it once per row. I'm not sure it's worth trying to fit this special case into our execution model just for testing.